timeseries/src/autogluon/timeseries/models/abstract/abstract_timeseries_model.py

```diff
@@ -273,39 +311,53 @@ def predict(
         """
         raise NotImplementedError
 
-    def predict_for_scoring(self, data: TimeSeriesDataFrame, **kwargs):
-        """Given a dataset, truncate the last `self.prediction_length` time steps and forecast these
-        steps with previous history. This method produces predictions for the *last* `self.prediction_length`
-        steps of the *given* time series, in order to be used for validation or scoring.
+    def slice_data_for_scoring(
```

Comment:
these two methods reside both here and in the trainer, and replicate the exact same logic. should they maybe move somewhere else and have only one implementation?

Reply:
1. Moved `slice_data_for_scoring` to `TimeSeriesDataFrame.get_model_inputs_for_scoring`
2. Moving `score_with_predictions` seems problematic:
    - We could only keep it as a method of `AbstractTimeSeriesModel`, but then inside the trainer we will need to load each model twice - once to predict & once to compute the score. This can be slow for local models that cache all predictions.
    - Could be moved to a static method in `utils`, but then will require additionally passing `target`, `prediction_length`, `metric` every time the method is invoked. 

     So I think it would be easier to keep `score_with_predictions` duplicated.